src/main/java/org/broadinstitute/barclay/argparser/CommandLineArgumentParser.java

```diff
@@ -1212,6 +1165,80 @@ private boolean hasRecommendedRange() {
             return this.maxRecommendedValue != Double.POSITIVE_INFINITY || this.minRecommendedValue != Double.NEGATIVE_INFINITY;
         }
 
+        // Validate the min/max range attribute values set in the code, and make sure
+        // any initial values are within range.
+        private void validateRangeDefinitions() {
+            if (hasBoundedRange() || hasRecommendedRange()) {
+                if (! Number.class.isAssignableFrom(this.type)) {
+                    throw new CommandLineException.CommandLineParserInternalException(
+                            String.format("Min/max value ranges can only be set for numeric arguments. "+
+                                    "Argument --%s has a minimum or maximum value but has a non-numeric type.",
+                                    this.getLongName()));
+                }
+                if (Integer.class.isAssignableFrom(this.type)) {
+                    if (!isInfinityOrMathematicalInteger(this.maxValue)
+                            || !isInfinityOrMathematicalInteger(this.minValue)
+                            || !isInfinityOrMathematicalInteger(this.maxRecommendedValue)
+                            || !isInfinityOrMathematicalInteger(this.minRecommendedValue)) {
+                        throw new CommandLineException.CommandLineParserInternalException(
```

Comment:
Can the maximum (or minimum) value be infinity? If so, this will fail.  I agree that users should not specify these values, but perhaps this error message should include the suggestion not to specify infinity.